### DISCUSSION TITLE
Dynamic route does not work on production

### DISCUSSION THREAD

**Discussion by AhmedNihmet** [vercel/next.js#28660] [Help]
### What version of Next.js are you using?

10.2.3

### What version of Node.js are you using?

0.1.0

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

Plesk - next start

### Describe the Bug

I have built a website with next js and somewhere on the process i had to add dynamic routes with( [getStaticPaths ](https://nextjs.org/docs/basic-features/data-fetching#getstaticpaths-static-generation)) based on my API to the website and everything worked as expected until i encountered this bug on **production** where my old route work just fine but if i where to add new routes(New projects) it will throw 404 and doesn't find it! so i have to go and rebuild the app completely and then restart it manual on the server and then the website updates with the newly added routes(Project route)!

Note: it works just fine when i run the application locally!

file structure :
Pages
      - projects/[id]

sample of the code:
![image](https://user-images.githubusercontent.com/51075069/131471833-c15dfca9-249a-4d56-aa77-48417c552852.png)

### Expected Behavior

New dynamic routes should be created when there are new projects in my API

### To Reproduce

see above.

**Comment by AlvesJorge** [comment]
Hello Ahmed,
I am no expert with nextjs but I understand that this is intended behavior.

Read the first few lines here on [GetStaticPaths](https://nextjs.org/docs/basic-features/data-fetching#getstaticpaths-static-generation)
"If a page has dynamic routes (documentation) and uses getStaticProps it needs to define a list of paths that have to be rendered to HTML at **build** time."

I think what would work for you is to use getServerSideProps, because these new pages are being generated after build time.
Hope this helps out.
PS. If anyone out there knows this better I would love to hear it!

**Comment by ijjk** [comment]
Hi, as mentioned by @AlvesJorge this is the expected behavior since you are using `fallback: false`, if you would like to generate pages for paths that were not returned from `getStaticPaths` during build-time you can leverage `fallback: true` (and handle the `useRouter().isFallback` case) or use `fallback: 'blocking'. 

Related documentation for the fallback behavior can be found here https://nextjs.org/docs/basic-features/data-fetching#the-fallback-key-required
